src/main/java/com/hivemq/cli/mqtt/AbstractMqttClientExecutor.java

```diff
@@ -123,6 +142,44 @@ public void subscribe(final @NotNull MqttClient client, final @NotNull Subscribe
         }
     }
 
+    @VisibleForTesting
+    @Nullable Tuple<MqttTopicFilter, MqttSharedTopicFilter> checkForSharedTopicDuplicate(
+            final @NotNull ClientData clientData, final @NotNull String topic) {
+        final Set<MqttTopicFilter> subscribedFilters = clientData.getSubscribedTopics();
+        final MqttTopicFilter newFilter = MqttTopicFilter.of(topic);
+        if (subscribedFilters.stream().anyMatch(MqttTopicFilter::isShared)) {
+            if (newFilter.isShared()) {
+                final Set<MqttTopicFilter> normalFilters =
+                        subscribedFilters.stream().filter(t -> !t.isShared()).collect(Collectors.toSet());
+                for (final MqttTopicFilter normalFilter : normalFilters) {
+                    final MqttSharedTopicFilter sharedFilter = ((MqttSharedTopicFilter) newFilter);
+                    if (IntersectionUtil.intersects(normalFilter, sharedFilter.getTopicFilter())) {
```

Comment:
The rest of the normalFilters could still be an intersection, right?
Is it correct to just return the tuple in this case?

Reply:
Well if it was decided that only the first intersection found counts, then this would be ok, but was it?

Reply:
Multiple intersection warnings are now possible.